app/recipeApp.js

```diff
@@ -0,0 +1,339 @@
+let recipeObject
+
+let recipes = []; 
+
+async function getData() {
+    const response = await fetch(
+        "https://raw.githubusercontent.com/jianxinz233/jianxinz233.github.io/refs/heads/main/data/recipe_app_data.json"
+    );
+    recipes  = await response.json();
+    showAllRecipes();
+}
+
+getData();
```

Comment:
an async function should be called with await. I don't see any await.
See https://dmitripavlutin.com/javascript-fetch-async-await/
and https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch